core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -94,6 +96,11 @@ public void initialize(String name, Map<String, String> properties) {
     closeableGroup.setSuppressCloseFailure(true);
   }
 
+  @Override
+  protected Map<String, String> properties() {
+    return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
```

Comment:
Depending on what you do above this may or may not be necessary

Reply:
I think this is still needed in case `initialize()` hasn't been called